### PR TITLE
UC9 Implementation

### PR DESCRIPTION
**Pull request by Jack-Is-2048**

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return inStoreSaleService.getAllInStoreSales();
+    }
+
+    /**
+     * Gets In Store Sales by Store ID
+     */
+    @GetMapping(path = "/store/{storeId}/sales")
+    public List<InStoreSale> getStoreSales(@PathVariable("storeId") Long storeId) {
+        return inStoreSaleService.getStoreSales(storeId);
+    }
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java:36]: Thinking more, mabye we should put this in the store controller but direct it to the inStoreSaleService?

File: src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * Add store to in store sale
+     */
+    @PostMapping(path = "/{inStoreSaleId}/store/{storeId}/add-store")
+    public InStoreSale addStoreToInStoreSale(@PathVariable("inStoreSaleId") Long inStoreSaleId,
+                                             @PathVariable("storeId") Long storeId) {
+        return inStoreSaleService.addStoreToInStoreSale(inStoreSaleId, storeId);
+    }
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java:53]: I would say this is not needed as we should include the store ID in the sale object when we initially create

File: src/main/java/com/mercuryCyclists/Sales/entity/Sale.java [hunk 3]
```diff
@@ -26,6 +27,7 @@ public abstract class Sale {
     private Long quantity;
 
     @Column(name = "date_time")
+    @JsonFormat(pattern="dd/MM/yy")
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/entity/Sale.java]: check my PR, probably better and easier to just set it to date.now()
happy with either way though tbh, but in reality we wouldn't let user set this

File: src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java (hunk 2 above)
Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java:53]: Can't test UC9 without creating sales, this was a temp solution, I agree though

File: src/main/java/com/mercuryCyclists/Sales/entity/Sale.java (hunk 3 above)
Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/Sales/entity/Sale.java]: I saw ur pr with the date.now() and I prefer it compared to how it is in this pr

File: src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java (hunk 2 above)
Comment (pedrooste) [reply src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java:53]: hmm actually this is tricky i think as you cant just add a store in the req body.

File: src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    /**
+     * Register a new Store
+     */
+    public InStoreSale registerInStoreSale(InStoreSale inStoreSale) {
+       if (!inStoreSale.validate()) {
+           throw new IllegalStateException("Invalid store");
+       }
+
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java:55]: you need to also check if the product is valid

File: src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        inStoreSaleObj.setStore(storeObj);
+        inStoreSaleRepository.save(inStoreSaleObj);
+        return inStoreSaleObj;
+    }
+
+    /**
+     * Updates existing in store sale based on the in store sale given
+     */
+    public InStoreSale updateInStoreSale(InStoreSale inStoreSale, Long inStoreSaleId) {
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java:85]: check product

File: src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java (hunk 1 above)
Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/Sales/controller/InStoreSaleController.java:36]: Seems a bit jank calling a diff service from a controller (store controller calling in store sale service)

**Pull request merged**